### DISCUSSION TITLE
1024 GB RAM

### DISCUSSION THREAD

**Discussion by valiDol42** [Chia-Network/bladebit#35] [Ideas]
I have 2 x Xeon(R) CPU E5-2673 v4 @ 2.30GHz and 1024 Gb RAM. Your plotter's best result was 511 sec. This is amazing! But we always want more... I tried to use all the RAM with two parallel processes (I turned off NUMA and cpu affinity and worked with NUMACTL), but this gave a result of more than 30 minutes each process. Is it possible to speed up the process using more memory than 416 GB?

**Comment by harold-b** [comment]
If you have 2 nodes only and 1024 RAM, you certainly should see some gains if you are plotting with memory specific to a single node. But, CPU affinity is overwritten in the application level currently so it likely overwrote your settings. 

I have a branch set up (not pushed yet) to make that optional. I'll alert you once that branch is tested and merged so that you can give it a try then. You should make sure that when you test it when the branch is merged, that you give divide actual physical cores evenly.

**Comment by valiDol42** [comment]
![QIP Shot - Screen 074](https://user-images.githubusercontent.com/19305835/127138441-2dcfcaf7-df7b-463a-a18b-c327af3d0263.png)
Thank you for answer and sorry for my English) 
I have corrected the original code to disable CPU affinity. 1 process running on 1 node (node0 - 40 cores, numactl --cpunodebind=0 --membind=0 and -m key for bladebit to disable Interleave NUMA mode) takes about 10 minutes.

I changed server configuration and now 2 parallel processes on 2 different nodes will be completed in 1000 sec (16-17 min) each!

![QIP Shot - Screen 077](https://user-images.githubusercontent.com/19305835/127145656-013d9028-bcba-49ee-8dcd-0f0081b5da17.png)
![QIP Shot - Screen 076](https://user-images.githubusercontent.com/19305835/127145658-e92fa3a9-1073-4f7a-96d8-30a57bcb25db.png)

**Reply by mocksoul** [reply]
But you said 511sec was with all cores and 1 process? So, no big improvements running this on parallel with double ram and separate numa nodes? Considering RAM prices it would be much better to buy extra server/cpu and move 512G ram there... ;)

**Reply by valiDol42** [reply]
Yeah... it was just an experiment. If the plotter could somehow use more memory to speed things up...
